packages/wrangler/src/user/user.tsx

```diff
@@ -1058,27 +1062,49 @@ export async function getAccountId(): Promise<string | undefined> {
 	const apiToken = getAPIToken();
 	if (!apiToken) return;
 
+	// check if we have a cached value
+	const cachedAccount = getConfigCache<{
+		account: { id: string; name: string };
+	}>("wrangler-account.json");
+	if (cachedAccount.account) {
+		return cachedAccount.account.id;
+	}
 	const accounts = await getAccountChoices();
 	if (accounts.length === 1) {
+		saveToConfigCache<{ account: { id: string; name: string } }>(
+			"wrangler-account.json",
+			{
+				account: { id: accounts[0].id, name: accounts[0].name },
+			}
+		);
```

Comment:
NIT: `saveToConfigCache()` is used four times in different files, which means there are multiple places to update if the shape of the data or the file path changes. Could we consider moving this into a helper function?